pkg/nakadi/nakadi.go

```diff
@@ -83,42 +167,63 @@ type statsPartition struct {
 // stats returns the Nakadi stats for a given subscription ID.
 //
 // https://nakadi.io/manual.html#/subscriptions/subscription_id/stats_get
-func (c *Client) stats(ctx context.Context, subscriptionID string) ([]statsEventType, error) {
+func (c *Client) stats(ctx context.Context, filter *SubscriptionFilter) ([]statsEventType, error) {
+	var subscriptionIDs []string
+	if filter.SubscriptionID == "" {
+		subscriptions, err := c.subscriptions(ctx, filter, "")
+		if err != nil {
+			return nil, err
+		}
+		subscriptionIDs = subscriptions
+	} else {
+		subscriptionIDs = []string{filter.SubscriptionID}
+	}
+
 	endpoint, err := url.Parse(c.nakadiEndpoint)
 	if err != nil {
 		return nil, err
 	}
 
-	endpoint.Path = fmt.Sprintf("/subscriptions/%s/stats", subscriptionID)
+	var stats []statsEventType
+	for _, subscriptionID := range subscriptionIDs {
+		endpoint.Path = fmt.Sprintf("/subscriptions/%s/stats", subscriptionID)
 
-	q := endpoint.Query()
-	q.Set("show_time_lag", "true")
-	endpoint.RawQuery = q.Encode()
+		q := endpoint.Query()
+		q.Set("show_time_lag", "true")
+		endpoint.RawQuery = q.Encode()
 
-	resp, err := c.http.Get(endpoint.String())
-	if err != nil {
-		return nil, err
-	}
-	defer resp.Body.Close()
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, endpoint.String(), nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create request: %w", err)
+		}
 
-	d, err := io.ReadAll(resp.Body)
-	if err != nil {
-		return nil, err
-	}
+		resp, err := c.http.Do(req)
+		if err != nil {
+			return nil, fmt.Errorf("failed to make request: %w", err)
+		}
+		defer resp.Body.Close()
 
-	if resp.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf("[nakadi stats] unexpected response code: %d (%s)", resp.StatusCode, string(d))
-	}
+		d, err := io.ReadAll(resp.Body)
+		if err != nil {
+			return nil, err
+		}
 
-	var result statsResp
-	err = json.Unmarshal(d, &result)
-	if err != nil {
-		return nil, err
-	}
+		if resp.StatusCode != http.StatusOK {
+			return nil, fmt.Errorf("[nakadi stats] unexpected response code: %d (%s)", resp.StatusCode, string(d))
```

Comment:
Same comment here, we should prepend the other error messages with `[nakadi stats]` prefix as well, I think.

Reply:
Added